Engine/Graphics/BSPModel.h

```diff
@@ -5,45 +5,56 @@
-#define FACE_TexAlignLeft       0x00001000
-#define FACE_IsInvisible        0x00002000
-#define FACE_TEXTURE_FRAME      0x00004000  // Texture ID is a frameset from TextureFrameTable, otherwise BitmapID
-#define FACE_TexAlignRight      0x00008000
-#define FACE_OUTLINED           0x00010000  // outline face debug
-#define FACE_TexAlignBottom     0x00020000
-#define FACE_TexMoveByDoor      0x00040000
-#define FACE_UNKOWN_10          0x00080000
-#define FACE_HAS_EVENT          0x00100000
-#define FACE_INDOOR_CARPET      0x00200000
-#define FACE_INDOOR_SKY         0x00400000
-#define FACE_FlipNormalU        0x00800000
-#define FACE_FlipNormalV        0x01000000
-#define FACE_CLICKABLE          0x02000000  // Event can be triggered by clicking on the facet.
-#define FACE_PRESSURE_PLATE     0x04000000  // Event can be triggered by stepping on the facet.
-#define FACE_INDICATE           0x06000000  // face has status bar string on hover
-#define FACE_TriggerByMonster   0x08000000
-#define FACE_TriggerByObject    0x10000000
-#define FACE_ETHEREAL           0x20000000  // Untouchable. You can pass through it.
-#define FACE_IsLava             0x40000000
-#define FACE_IsPicked           0x80000000  // mouse is hovering
+#include "src/tools/Flags.h"
+
+enum class FaceAttribute : uint32_t {
+    FACE_IsPortal          = 0x00000001,
+    FACE_IsSecret          = 0x00000002,
+    FACE_FlowDown          = 0x00000004,
+    FACE_TexAlignDown      = 0x00000008,
+    FACE_IsFluid           = 0x00000010,
+    FACE_FlowUp            = 0x00000020,
+    FACE_FlowLeft          = 0x00000040,
+    FACE_SeenByParty       = 0x00000080,
+    FACE_XY_PLANE          = 0x00000100,
+    FACE_XZ_PLANE          = 0x00000200,
+    FACE_YZ_PLANE          = 0x00000400,
+    FACE_FlowRight         = 0x00000800,
+    FACE_TexAlignLeft      = 0x00001000,
+    FACE_IsInvisible       = 0x00002000,
+    FACE_TEXTURE_FRAME     = 0x00004000,  // Texture ID is a frameset from TextureFrameTable, otherwise BitmapID
+    FACE_TexAlignRight     = 0x00008000,
+    FACE_OUTLINED          = 0x00010000,  // outline face debug
+    FACE_TexAlignBottom    = 0x00020000,
+    FACE_TexMoveByDoor     = 0x00040000,
+    FACE_UNKOWN_10         = 0x00080000,
+    FACE_HAS_EVENT         = 0x00100000,
+    FACE_INDOOR_CARPET     = 0x00200000,
+    FACE_INDOOR_SKY        = 0x00400000,
+    FACE_FlipNormalU       = 0x00800000,
+    FACE_FlipNormalV       = 0x01000000,
+    FACE_CLICKABLE         = 0x02000000,  // Event can be triggered by clicking on the facet.
+    FACE_PRESSURE_PLATE    = 0x04000000,  // Event can be triggered by stepping on the facet.
+    FACE_INDICATE          = 0x06000000,  // face has status bar string on hover
+    FACE_TriggerByMonster  = 0x08000000,
+    FACE_TriggerByObject   = 0x10000000,
+    FACE_ETHEREAL          = 0x20000000,  // Untouchable. You can pass through it.
+    FACE_IsLava            = 0x40000000,
+    FACE_IsPicked          = 0x80000000,  // mouse is hovering
+};
```

Comment:
Significant differences in names from MMExtension [here](https://github.com/GrayFace/MMExtension/blob/74e8d73db58e1f5cfdfc4376578729935c3b44aa/Scripts/Core/ConstAndBits.lua#L53):
- FACE_UNKOWN_10 ↔ TriggerByTouch = 0x00080000,  -- doesn't work anymore
- FACE_HAS_EVENT ↔ IsEventJustHint = 0x00100000,  -- [MM7+]
- FACE_INDOOR_CARPET ↔ AlternativeSound = 0x00200000,
- FACE_IsPicked ↔ HasData = 0x80000000, -- that one is set on deserialized data during load on many indoor faces/outdoor models and caused brief picked debug flashing bug on map enter. Maybe it means it have [this data](https://github.com/GrayFace/MMExtension/blob/74e8d73db58e1f5cfdfc4376578729935c3b44aa/Scripts/Structs/01%20common%20structs.lua#L1813) (we are named this BLVFaceExtra, fields of which is mostly unnamed).